Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1000,6 +1000,20 @@ public long GetNextTxNonce(Address address)
                 {
                     _rwlock.ExitWriteLock();
                 }
+
+                // Prepare TxExecutions.  Note that these should be realized into an actual array
+                // before _rwlock.EnterWriteLock(), because the following function tries to access
+                // to the current states.
+                TxExecution[] txExecutions = MakeTxExecutions(block, actionEvaluations).ToArray();
```

Comment:
How about unifying `MakeTxExecution()` calls here and the inside of `Append()`? If `TxExecution` is of a type similar to `ActionEvaluation`, it might be treated similarly. (Calculate once and store only once)